src/imap.rs

```diff
@@ -446,28 +448,37 @@ impl Imap {
                     err
                 );
 
-                return 0;
+                return false;
             }
         };
 
         self.config.write().unwrap().should_reconnect = false;
 
         match login_res {
             Ok((session, stream)) => {
-                *self.session.lock().unwrap() = (Some(session), Some(stream));
-                1
+                *self.session.lock().unwrap() = Some(session);
+                *self.stream.write().unwrap() = Some(stream);
```

Comment:
why this change?

Reply:
huh? that was your commit from imap-locks branch that i incorporated here ;)